### ISSUE TITLE
SPA implementation issue

### ISSUE THREAD

**Issue by jdhrnndz** [sinnerschrader/aem-react#40]
The doc says that the page structure for the SPA must match the routes defined. I used my component with routes as template for the parent page of the SPA however I cannot make the routing work because I am unsure what template should I use for the children pages. Currently, the root page has a template of the component where the routes are defined, and it renders properly.

**Comment by stemey** [comment]
Each page has to have the router component in the same path relative to its root. So in the cityfinder example the cities page has ist under [jcr:content/par/city_finder](//github.com/sinnerschrader/aem-react/blob/master/demo/src/main/content/jcr_root/content/reactdemo/cities/.content.xml#L14).
And the subpages as well: [hamburg](//github.com/sinnerschrader/aem-react/blob/master/demo/src/main/content/jcr_root/content/reactdemo/cities/hamburg/.content.xml#L14)

To make it possible to drop the cityFinder anywhere it locates the root route by looking at the depth property. 

Hope that helps

**Comment by jdhrnndz** [comment]
Yes, **this did solve the issue**. For anyone referring to this, the problem without the `depth` attribute is that when accessing the root page, it goes one level higher instead of going to the intended path. For example I have this:

```
my-project
    - countries
        - usa
        - canada
```

Without setting the CountryFinder's `depth` to 0, accessing `my-project/countries.html` will match no routes because it would resolve to `my-project.html`. 

TL;DR - Set the pages' `depth` attribute accordingly when using the SPA implementation.

**Comment by jdhrnndz** [comment]
Hello. I have encountered another problem in implementing SPA.

I have followed your example at aem-react-demo. The routes worked as expected except that the content cannot be fetched and instead returns the error that says: 
`Error: cannot find component for resourceType components/content/react/dashboard-section-view-component`

However, it renders correctly when I refresh the page. What could be the mistakes that I need to fix?
